### PR TITLE
Problem: want to install additional config files

### PR DESCRIPTION
**Pull request by vyskocilm**
Solution: add type="config" to zproject_install.gsl

### CODE REVIEW

File: zproject_install.gsl [hunk 1]
```diff
@@ -22,6 +22,7 @@ function zinstall_include (filename)
     endif
 endfunction
 
+directory.create ("builds/zinstall")
```
Comment (sappo) [inline zproject_install.gsl]: Should only be created if zinstall is used!

**Comment by sappo** [comment]
I'm not sure this problem is valid. Configuration files should never be part of the project itself! What are those configuration files?

**Comment by sappo** [comment]
@vyskocilm do you have an example project?

File: zproject_install.gsl (hunk 1 above)
Comment (vyskocilm) [inline zproject_install.gsl]: I tried the simplest way how to do it
project.has_install = (count (main, (count (main.install) > 0)) > 0)

but it crashes gsl. I'll create minimal reproducer and add issue for imatix/gsl.

Can we solve it in another PR, please?

**Comment by vyskocilm** [comment]
@sappo https://github.com/42ity/fty-metric-composite/blob/master/src/fty-metric-composite.cfg.example

The example configuration is treated as a template for various instances of other services. Basically what can be changed is to drop type="config" and create a type="datadir". Makes a sense for you?

**Comment by jimklimov** [comment]
I guess this installation of configuration files refers to sample configs
and/or those generated from templates (like autotools .in support) during
build for a particular distribution with particular FS layout (if that is
involved/specified in the config file), user account names etc. - things
that are not hardcoded in project sources and imposed on all distros, but
that can well be automated to simplify secure setup and working out of the
box on the final system.


2017-01-12 11:01 GMT+01:00 Kevin Sapper <notifications@github.com>:

File: zproject_install.gsl (hunk 1 above)
Comment (sappo) [inline zproject_install.gsl]: A simple solution might be:
```
for project.main where defined (main->install)
    if !file.exists ("builds/zinstall")
        directory.create ("builds/zinstall")
    endif
...
```

**Comment by sappo** [comment]
I understand the problem, still I think a user wont get the model. But lacking any better ideas I'll merge it for now.

**Comment by sappo** [comment]
@vyskocilm #834 might be a cleaner solution.

**Pull request merged**